src/demo/share/jfc/SwingSet2/SwingSet2.java

```diff
@@ -848,6 +848,11 @@ private void updateThisSwingSet() {
         if (frame == null) {
             SwingUtilities.updateComponentTreeUI(this);
         } else {
+            if (currentLookAndFeel.name.contains("GTK")) {
+                frame.setSize(980, 870);
```

Comment:
why do you update the size here instead of above where the original code sets the preferred size of the demo?

Reply:
Preferred size is set only when the demo is initialized and it remains of the same size for all the L&F and that results in clipping of few demos for GTK L&F. The solution is to change the preferred width and height for all the L&F as it was done in the initial fix but @prsadhuk suggested [here](https://github.com/openjdk/jdk/pull/20052#issuecomment-2217287962) to change the size only for GTK L&F as frame size for other L&F doesn't clip the demos.

So, whenever the L&F is set to GTK, the frame size is modified to handle the clipping and restore the original size if the L&F is set to other than GTK.